Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -313,9 +324,15 @@ internal static string DeterminePrimaryScriptFile(JObject functionConfig, string
                         Path.GetFileNameWithoutExtension(p).ToLowerInvariant() == "run" ||
                         Path.GetFileName(p).ToLowerInvariant() == "index.js");
                 }
+            }
 
-                return functionPrimary;
+            if (string.IsNullOrEmpty(functionPrimary))
+            {
+                throw new ConfigurationErrorsException("Unable to determine the primary function script. Try renaming your entry point script to 'run' (or 'index' in the case of Node), " +
+                    "or alternatively you can specify the name of the entry point script explicitly by adding a 'scriptFile' property to your function metadata.");
```

Comment:
You throw various exception with descriptive messages, but don't they all get ignored in `TryGetFunctionConfigAsync`? Is so, what's the point?